tests/unit/src/main/java/org/mule/tck/core/internal/serialization/CustomByteArrayInputStream.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tck.core.internal.serialization;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CustomByteArrayInputStream extends InputStream {
```

Comment:
Use a more declarative name for the class. This is not necessarily a ByteInputStream that is decorated, it is a wrapper/decorator for a InputStream.

Something like CloseCounterInputStreamWrapper can be an option.

Add a jdoc indicating the driver behind this (avoid spy for java 17)